cli/docker/environment_source.go

```diff
@@ -0,0 +1,56 @@
+package docker
+
+import (
+	"os"
+	"strings"
+
+	"github.com/bitrise-io/bitrise/log"
+)
+
+// implementing env.EnvironmentSource
+type DockerEnvironmentSource struct {
+	Logger log.Logger
+}
+
+func (des *DockerEnvironmentSource) GetEnvironment() map[string]string {
+	passthroughEnvsList := strings.Split(os.Getenv("BITRISE_DOCKER_PASSTHROUGH_ENVS"), ",")
```

Comment:
I don't see it defined anywhere else, how does this work?
Note: there is a file `configs.go` for collecting these sort of constants, it should probably live there.

Reply:
At this point this is an internal undocumented "API" used by DEN to supply the name of the environment variables that should be passed through automatically to the container environments. Sort of a dynamic allowlist. This prevents passing through environment variables that would potentionally break containers, such as JAVA_HOME, PWD, etc. But all Bitrise relevant envs are allowed.